### PR TITLE
Add pre-commit and lint

### PR DESCRIPTION
**Pull request by jianyunchao**
Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:

- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [x] You have added unit tests

## Motivation

(Write your motivation for proposed changes here.)

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+### Fix Bugs
+
+Look through the GitHub issues for bugs. Anything tagged with "bug" and "help
+wanted" is open to whoever wants to implement it.
+
+### Implement Features
+
+Look through the GitHub issues for features. Anything tagged with "enhancement"
+and "help wanted" is open to whoever wants to implement it.
```
Comment (HaoyangLee) [inline CONTRIBUTING.md:32]:
```suggestion
and "help wanted" is open to whoever wants to fix it.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+4. Create a branch for local development:
+
+   ```shell
+   git checkout -b name-of-your-bugfix-or-feature
+   ```
+
+   Now you can make your changes locally.
+
+5. When you're done making changes, check that your changes pass the linters and the tests:
```
Comment (HaoyangLee) [inline CONTRIBUTING.md:84]:
```suggestion
5. When you finish making changes, check that your changes pass the linters and the tests:
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+```shell
+pre-commit install
+```
+
+now `pre-commit` will run automatically on `git commit`!
+
+## Releasing
+
+A reminder for the maintainers on how to deploy.
+Make sure all your changes are committed (including an entry in HISTORY.md).
```
Comment (HaoyangLee) [inline CONTRIBUTING.md:138]:
```suggestion
Make sure all your changes are committed.
```

Comment (HaoyangLee) [reply CONTRIBUTING.md:138]: remove if unnecessary

**Review by HaoyangLee — APPROVED** [review]

**Review by SamitHuang — APPROVED** [review]

**Pull request merged**